### PR TITLE
notebook tests need to use a unique subdirectory

### PR DESCRIPTION
**Pull request by jlewi**
* When the notebook runs on a kubeflow-ci-deployment cluster it will
  upload the rendered notebook to a gcs bucket in kubeflow-ci-deployment.

* We need the GCS path for the notebook to be unique so that results
  from different runs don't overwrite each other.

Related to: #715

### CODE REVIEW

File: py/kubeflow/testing/notebook_tests/execute_notebook.py [hunk 1]
```diff
@@ -62,6 +64,15 @@ def run_notebook_test(notebook_path, parameters=None):
   html_exporter = nbconvert.HTMLExporter()
   (html_output, _) = html_exporter.from_notebook_node(nb)
   gcs_path = os.getenv("OUTPUT_GCS")
+
+  # Per https://github.com/kubeflow/testing/issues/715
+  # we need to add some uniquness to the name since different test runs
+  # will use the same OUTPUT_GCS directory
+  subdir = datetime.datetime.now().strftime("%Y%m%d-%H%M")
+  subdir = "-" + uuid.uuid4().hex[0:4]
```
Comment (NikeNano) [inline py/kubeflow/testing/notebook_tests/execute_notebook.py:72]: Do we wish to combine the date and the uuid? The date is just overwritten.

Comment (jlewi) [reply py/kubeflow/testing/notebook_tests/execute_notebook.py]: @NikeNano Good catch thanks; it was a bug.

Comment (jlewi) [reply py/kubeflow/testing/notebook_tests/execute_notebook.py]: Fixed. @NikeNano PTAL

**Pull request merged**